v2/dtos/requests/event.go

```diff
@@ -89,3 +89,14 @@ func AddEventReqToEventModel(addEventReq AddEventRequest) (event models.Event) {
 		Tags:        tags,
 	}
 }
+
+func NewAddEventRequest(requestId string, event dtos.Event) AddEventRequest {
+	event.Versionable = common.NewVersionable()
+	for _, reading := range event.Readings {
+		reading.Versionable = common.NewVersionable()
```

Comment:
In this case `reading` is a copy, so you must use the `index` to modify properties on the reading for change to stick outside the current scope of `reading`

Reply:
Thank you for catching that!   Modified.